KSR_002/KSR_002.md

```diff
@@ -0,0 +1,282 @@
+
+`start_stack `contains the start address of the stack
+
+`struct vm_area_struct * mmap`  is used to identify the all the VMA regions mapped in the current address space ( <mark > **TODO:** </mark> check if there is any mapped area in the telltale safety app)
+
+
+## b. 
+The Kernel is responsible for maintaining the integrity of process descriptors, process address spaces and associated structures (memory management structures) at runtime when memory is allocated, used and freed. 
+
+<mark > **TODO:** </mark> investigate what happens upon memory mapping, allocation and free.
+
+
+### Initial Considerations
+Assuming that the heap area is not critical in the context of the [telltale safety app](https://github.com/elisa-tech/wg-automotive-safety-app/blob/main/safety-app.c) the initial investigation should be focused on the maintenance of the code and data sections as well as the stack section. More specifically: 
+- how are these areas maintained during the execution of user code?
+- how can they be corrupted?
+	 
+## c. 
+The Kernel is responsible for maintaining the integrity of process descriptors and process address spaces at runtime when doing context switches. There are multiple situations that can generate a context switch:
+- syscalls: e.g. for x86 [entry_SYSCALL_64](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/x86/entry/entry_64.S?h=v5.14#n87);
+- explicit invocation of the scheduler __schedule();
+- Interrupt and Exception handling
+- Signals being sent between processes (<mark > **TODO:** </mark> investigate further what can happen here)  
+context_switch() show user space mm being borrowed by kernel threads upon context switch:  
+<mark > **TODO:** </mark> understand why this is required and if we can use a dummy mm instead of borrowing a real user space mm  
+<mark > **TODO: (possibly in a different investigation thread)** </mark>: investigate the role of membarrier_switch_mm() when switching to user thread
+
+
+## d. 
+The Kernel subsystems and the HW supporting the integrity of processes address spaces upon creation and at runtime shall be properly configured at boot time. This is responsibility of start_kernel(); <mark > **TODO:** </mark> identify the subsystems of the Kernel that following the start_kernel() are invoked to meet this requirement  
+<mark > **TODO:** </mark> Investigate if init user space processes (e.g. systemd) can impact the capability/functionalities of Kernel subsystems responsible for guaranteeing the integrity of process address spaces during creation or runtime.
+
+
+# NOTE: 
+
+**Previous material relevant to this topic:**
```

Comment:
note: for now we can leave the text below but maybe later on we need to remove as this material should go into the lfscs osep wg githubs